### PR TITLE
docs: add layered configuration

### PR DESCRIPTION
**Pull request by zyy17**

### CODE REVIEW

File: docs/user-guide/operations/configuration.md [hunk 1]
```diff
@@ -338,3 +338,44 @@ datanode_lease_secs = 30
...
+
+- Command line flags
+- Configuration file
+- Environment variables
+- Default values
+
+Every item in the configuration file can be mapped into environment variables. For example, if we want to set the configuration item `max_inflight_tasks ` of datanode by environment variables:
+
+```toml
+...
```
Comment (sunng87) [inline docs/user-guide/operations/configuration.md:354]:
```suggestion
# ...
```

File: docs/user-guide/operations/configuration.md [hunk 2]
```diff
@@ -338,3 +338,44 @@ datanode_lease_secs = 30
...
+[storage.compaction]
+# Max task number that can concurrently run.
+max_inflight_tasks = 4
+...
```
Comment (sunng87) [inline docs/user-guide/operations/configuration.md:358]:
```suggestion
# ...
```

File: docs/user-guide/operations/configuration.md [hunk 3]
```diff
@@ -338,3 +338,44 @@ datanode_lease_secs = 30
...
+
+```toml
+# ...
+[storage.compaction]
+# Max task number that can concurrently run.
+max_inflight_tasks = 4
+# ...
+```
+
+You can use the following format:
```
Comment (nicecui) [inline docs/user-guide/operations/configuration.md:361]:
```suggestion
You can also use the following format in your own environment file:
```

I guess this is for users' own environment file, is it right?

**Review by nicecui — COMMENTED** [review]
Could you please add the following things:

1. how to config command line flags?
2. list all the configurations.

File: docs/user-guide/operations/configuration.md (hunk 3 above)
Comment (zyy17) [reply docs/user-guide/operations/configuration.md]: @nicecui The current implementation doesn't read environment variables from the file. We can use the following shell command to setup the environment variables for example:

```shell
$ export GREPTIMEDB_DATANODE__STORAGE__COMPACTION__MAX_INFLIGHT_TASKS=4
$ echo $GREPTIMEDB_DATANODE__STORAGE__COMPACTION__MAX_INFLIGHT_TASKS
4
```

I already update the docs to add more details.

**Comment by zyy17** [comment]
> Could you please add the following things:
> 
> 1. how to config command line flags?
> 2. list all the configurations.

Sounds reasonable, I'll add it later.

**Review by nicecui — APPROVED** [review]
LGTM

File: docs/user-guide/operations/configuration.md [hunk 4]
```diff
@@ -1,21 +1,100 @@
...
+- `--grpc-addr <GPRC_ADDR>`: GRPC server address;
+- `--metasrv-addr <METASRV_ADDR>`: Metasrv address list;
+- `--mysql-addr <MYSQL_ADDR>`: MySQL server address;
+- ` --opentsdb-addr <OPENTSDB_ADDR>`: OpenTSDB server address;
+- `--postgres-addr <POSTGRES_ADDR>`: Postgres server address;
+- `--prom-addr <PROM_ADDR>`: Prometheus server address;
+- `--tls-cert-path <TLS_CERT_PATH>`: The TLS public key file path;
+- `--tls-key-path <TLS_KEY_PATH>`: The TLS private key file path;
+- `--tls-mode <TLS_MODE>`: TLS Mode;
+- `--user-provider <USER_PROVIDER>`: You can refer [authentication](https://docs.greptime.com/user-guide/clients/authentication);
```
Comment (sunng87) [inline docs/user-guide/operations/configuration.md:85]: Consider using relative path for link

```suggestion
- `--user-provider <USER_PROVIDER>`: You can refer [authentication](/user-guide/clients/authentication);
```

File: docs/user-guide/operations/configuration.md [hunk 5]
```diff
@@ -1,21 +1,100 @@
...
+```
+greptime metasrv start --help
+```
+
+- `-c`/`--config-file`: The configuration file for metasrv;
+- `--env-prefix <ENV_PREFIX>`: The prefix of environment variables, default is `GREPTIMEDB_METASRV`;
+- `--bind-addr <BIND_ADDR>`: The bind address of metasrv;
+- `--http-addr <HTTP_ADDR>`: HTTP server address;
+- `--http-timeout <HTTP_TIMEOUT>`: HTTP request timeout;
+- `--selector <SELECTOR>`: You can refer [selector-type](https://docs.greptime.com/developer-guide/meta/selector#selector-type);
```
Comment (sunng87) [inline docs/user-guide/operations/configuration.md:57]:
```suggestion
- `--selector <SELECTOR>`: You can refer [selector-type](/developer-guide/meta/selector#selector-type);
```

**Review by sunng87 — APPROVED** [review]

**Pull request merged**